src/dataframerow/utils.jl

```diff
@@ -67,7 +83,12 @@ function hashrows(cols::Tuple{Vararg{AbstractVector}}, skipmissing::Bool)
     rhashes = zeros(UInt, len)
     missings = fill(false, skipmissing ? len : 0)
     for (i, col) in enumerate(cols)
-        hashrows_col!(rhashes, missings, col, i == 1)
+        rp = DataAPI.refpool(x)
+        if rp === nothing
+            hashrows_col!(rhashes, missings, col, i == 1)
+        else
+            hashrows_col_pool!(rhashes, missings, col, rp, i == 1)
+        end
```

Comment:
I'd typically rely on dispatch for that: `hashrows_col_pool!` can be a method of `hashrows_col!` when the pool is `AbstractVector` (or even `Any` given my comment above, with `Nothing` being caught by the fallback).

Reply:
ok